src/components/featured-products-section/featured-products-section.tsx

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from 'react';
+import { ProductCard } from '../product-card/product-card';
+import { ProductLink } from '../product-link/product-link';
+import styles from './featured-products-section.module.scss';
+import { collections } from '@wix/stores';
+import { getEcomApi } from '~/api/ecom-api';
+import { Product } from '@wix/stores_products';
+import classNames from 'classnames';
+
+interface FeaturedProductsData {
+    category: collections.Collection;
+    products: Product[];
+}
+
+const getCategoryProducts = async (
+    categorySlug: string,
+    limit: number
+): Promise<FeaturedProductsData | null> => {
+    const api = getEcomApi();
+
+    let category: collections.Collection;
+    try {
+        category = await api.getCategoryBySlug(categorySlug);
+    } catch (error) {
+        try {
+            category = await api.getCategoryBySlug('all-products');
+        } catch (error) {
+            return null;
+        }
+    }
+
+    try {
+        const products = await api.getFeaturedProducts(category.slug, limit);
+        return { category, products };
+    } catch (error) {
+        return null;
```

Comment:
I'd rather don't show the entire section in error case. For now we show title and description.